sw/device/lib/base/memory.c

```diff
@@ -45,12 +130,37 @@ enum {
 OT_WEAK
 int OT_PREFIX_IF_NOT_RV32(memcmp)(const void *lhs, const void *rhs,
                                   size_t len) {
-  const uint8_t *lhs8 = (uint8_t *)lhs;
-  const uint8_t *rhs8 = (uint8_t *)rhs;
-  for (size_t i = 0; i < len; ++i) {
-    if (lhs8[i] < rhs8[i]) {
+  if (len == 0) {
+    return kMemCmpEq;
+  }
```

Comment:
nit: wdyt about removing this? This doesn't seem like a case we would encounter a lot.

Reply:
Done